bindsnet/network/topology.py

```diff
@@ -148,12 +148,12 @@ def __init__(self, source: Nodes, target: Nodes, nu: Optional[Union[float, Seque
 
         self.w = kwargs.get('w', None)
         if self.w is None:
-            if self.wmin is None or self.wmax is None:
+            if self.wmin == -np.inf or self.wmax == np.inf:
                 self.w = torch.rand(source.n, target.n)
-            elif self.wmin is not None and self.wmax is not None:
+            else:
                 self.w = self.wmin + torch.rand(source.n, target.n) * (self.wmax - self.wmin)
         else:
-            if self.wmin is not None and self.wmax is not None:
+            if self.wmin != -np.inf and self.wmax != np.inf:
```

Comment:
What is one is finite? E.g, if `self.wmin = -np.inf` and `self.wmax=1`, this statement won't execute. Should be an `or` clause.